common/src/main/java/com/lightstep/opentelemetry/common/VariablesConverter.java

```diff
@@ -41,9 +45,12 @@ public static void setSystemProperties(Configuration configuration, boolean isAg
 
     if (!configuration.spanEndpoint.toLowerCase().startsWith("http://")
         && !configuration.spanEndpoint.toLowerCase().startsWith("https://")) {
-      String msg = "Invalid configuration: span endpoint should start with http:// or https://.";
-      logger.severe(msg);
-      throw new IllegalStateException(msg);
+
+      if (configuration.insecureTransport) {
+        configuration.spanEndpoint = "http://" + configuration.spanEndpoint;
```

Comment:
Let's do a pass in a follow-up PR (in the future), saving these constants (`http`, `https`, etc) in a `Constants.java` file ;)

Reply:
ok